app/models/service.rb

```diff
@@ -192,4 +195,13 @@ def unapproved_changes?(attribute, meh=false)
     end
   end
 
+  def publicly_visible?
+    visible && discarded_at.nil?
+  end
+
+  def self.convert_time_fields(service_object)
```

Comment:
Not really happy with this being in the Service model, but not sure where else to put this, as it's a Service/Snapshot related bit of work.  Since there is upcoming work to control what JSON is emitted, public fields etc. for the API, may move this to a new package for that

Reply:
An explanation of why it is needed.
These fields are Ruby date objects, not DateTimes, so their JSON representation is "YYYY-MM-DD".  When this is put into Mongo verbatim, date queries, such as $gte don't work, so this adds a time and zone component to it.